### PR TITLE
travis: Adding travis support

### PR DESCRIPTION
**Pull request by ErwanAliasr1**

### CODE REVIEW

**Comment by ErwanAliasr1** [comment]
@leseb It's green on travis ! LGTM

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - mv $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nat $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nonat
+  - make
+  - sudo ./cn version
+  - sudo make tests
+
+after_failure:
+  - docker images
+  - docker ps
+  - docker ps -a
+  - docker logs
```
Comment (leseb) [inline .travis.yml]: please remove

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+script:
+  - make prepare
+  - mv $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nat $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nonat
+  - make
+  - sudo ./cn version
+  - sudo make tests
+
+after_failure:
+  - docker images
+  - docker ps
```
Comment (leseb) [inline .travis.yml]: please remove

File: tests/functional-tests.sh [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  echo $filename
+}
+
+function deleteFile() {
+  if [ -e "$1" ]; then
+    rm -f $1 || fatal "Cannot delete $1"
+  fi
+}
+
+function run_cn() {
```
Comment (leseb) [inline tests/functional-tests.sh]: `runCn`

File: tests/functional-tests.sh [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+function update {
+  run_cn update
+}
+
+function version {
+  run_cn version
+}
+
+function s3_mb {
```
Comment (leseb) [inline tests/functional-tests.sh]: s3Mb

**Review by leseb — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (ErwanAliasr1) [reply .travis.yml]: done

File: .travis.yml (hunk 1 above)
Comment (ErwanAliasr1) [reply .travis.yml]: done

File: tests/functional-tests.sh (hunk 3 above)
Comment (ErwanAliasr1) [reply tests/functional-tests.sh]: done

File: tests/functional-tests.sh (hunk 4 above)
Comment (ErwanAliasr1) [reply tests/functional-tests.sh]: I suggest we switch to test_<something>

Comment (leseb) [reply tests/functional-tests.sh]: let's do this

**Review by leseb — APPROVED** [review]

**Comment by ErwanAliasr1** [comment]
This seems good to me, please merge if it's ok for you. I'll open a new PR to increase the test's complexity

File: cmd/s3_del.go [hunk 5]
```diff
@@ -14,8 +14,8 @@ var (
 // CliS3CmdDel is the Cobra CLI call
 func CliS3CmdDel() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "del BUCKET/OBJECT",
-		Short: "Delete bucket",
+		Use:   "del BUCKET/FILE",
```
Comment (leseb) [inline cmd/s3_del.go]: OBJECT was correct.

File: cmd/s3_del.go [hunk 6]
```diff
@@ -14,8 +14,8 @@ var (
...
+		Short: "Delete file from bucket",
```
Comment (leseb) [inline cmd/s3_del.go]: We should test this, does `del` link to `rb` if the bucket is empty?

**Review by leseb — CHANGES_REQUESTED** [review]

File: cmd/s3_del.go (hunk 6 above)
Comment (ErwanAliasr1) [reply cmd/s3_del.go]: I added the test for del

I didn't checked the behavior in an empty bucket. I would be more logical not deleting the bucket.

File: cmd/s3_del.go (hunk 5 above)
Comment (ErwanAliasr1) [reply cmd/s3_del.go]: reverted.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+script:
+  - make prepare
+  - mv $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nat $GOPATH/src/github.com/docker/docker/vendor/github.com/docker/go-connections/nonat
+  - make
+  - sudo ./cn version
+  - sudo make tests
+
+after_script:
+  - ./cn stop
+  - ./cn purge
```
Comment (leseb) [inline .travis.yml]: `./cn purge --yes-i-am-sure`.

**Review by leseb — CHANGES_REQUESTED** [review]
last thing

File: .travis.yml (hunk 7 above)
Comment (ErwanAliasr1) [reply .travis.yml]: good catch

Comment (ErwanAliasr1) [reply .travis.yml]: repushed

**Review by leseb — APPROVED** [review]

**Pull request merged**